src/coreclr/jit/compiler.cpp

```diff
@@ -4221,6 +4221,65 @@ bool Compiler::compRsvdRegCheck(FrameLayoutState curState)
 }
 #endif // TARGET_ARMARCH || TARGET_RISCV64
 
+//------------------------------------------------------------------------
+// FindParameterRegisterLocalMappingByRegister:
+//   Try to find a mapping that maps a particular parameter register to an
+//   incoming defined local.
+//
+// Returns:
+//   The mapping, or nullptr if no mapping was found for this register.
+//
+const ParameterRegisterLocalMapping* Compiler::FindParameterRegisterLocalMappingByRegister(regNumber reg)
+{
+    if (m_paramRegLocalMappings == nullptr)
+    {
+        return nullptr;
+    }
+
+    for (int i = 0; i < m_paramRegLocalMappings->Height(); i++)
+    {
+        const ParameterRegisterLocalMapping& mapping = m_paramRegLocalMappings->BottomRef(i);
```

Comment:
i am assuming an `ArrayStack` is used  instead of map to keep things simpler and there are not many entries in the `ArrayStack` that this will cause performance issues?